src/backends/perf_pt/mod.rs

```diff
@@ -406,6 +406,18 @@ impl PerfPTTracer {
             return Err(HWTracerError::NoHWSupport("Intel PT not supported by CPU".into()));
         }
 
+        // Check for inavlid configuration.
+        fn power_of_2(v: size_t) -> bool {
+            !(v <= 0) && ((v & (v - 1)) == 0)
+        }
+        if !power_of_2(config.data_bufsize) {
+            return Err(HWTracerError::BadConfig(String::from("data_bufsize must be a positive power of 2")));
```

Comment:
If we're going to have a `power_of_2` function (which seems reasonable), can't we chuck most of this boilerplate in there too?

Reply:
Sure.

I also considered moving it into `util.rs` as it's something we might need in other places later. In that case we don't want the error handling in there.

Thoughts?

Reply:
I'm fine with having it here  -- if it's not used elsewhere, there's no advantage to moving it.

Reply:
Wait, we need to raise two different error messages for the two call sites, so I think we probably want to leave this as-is.